### ISSUE TITLE
Flans Mod vehicles not taking damage from vanilla and other mods

### ISSUE THREAD

**Issue by SnowyJune678** [FlansMods/FlansMod#1247]
**Describe the solution you'd like**
I would like for Flans Mod vehicles and their occupants to be damageable by non-Flans items and blocks. They appear to be invincible from vanilla and other mods including:
-Techguns guided missiles
-ICBM Classic nuclear explosives
-Trinity (Nuclearcraft addon) nuclear explosives, including to a certain extent, from the fallout
-Vanilla lava (including literally driving through Nether's lava seas)
-Vanilla TNT
-The Wither
-Ice and Fire dragons (though the dragons did throw me out of the vehicle before killing me)

Most of the testing has been done on the M1A1 Abrams.

**Describe alternatives you've considered**
There aren't good alternative to Flans Mod imo.

**Additional context**
none

**Comment by JValor2417** [comment]
Hey @0something0, 

I know the issues section can take quite some time to be looked through and resolved or answered, so I’ll give my two cents. 

Firstly, I am pretty sure the damage from most other mods/vanilla not working on flans mod vehicles is due to flans mod’s health system being something far different. I resolved this on my server by using custom NPCs of the mobs I wanted. they fired custom NPCs projectiles in all different ways relative to how I needed them to act,  ( I used custom NPC’s because I found that for whatever reason those projectiles can damage the vehicles, whilst keeping what is essentially the mobs you’d needed. ) So this is a possible workaround. 

Second, I am very sure fire set to flans mod vehicles deals a hefty amount of damage. You said you tested this on a tank, which could explain maybe why barely-to-no damage showed from the lava? From my experience doing that in any weak vehicle would burn it and then the people/cargo inside. This could all tie into the flans mod health system, too. 

In conclusion, until the official team works on this, I’d use something like custom NPC’s and recreate the mobs with it how you’d need, and maybe get them to deal damage in their attributes tab. 

P.s I think arrows can deal damage to vehicles too? It’s mostly noticeable on weaker vehicles from what I remember.

**Comment by Jrd458** [comment]
Just wondering how did you get the customer NPC's to damage the vehicle when I try the projectiles hit the flans vehicles but dealt no damage to the vehicle itself
